runtime/verified_gc/gc.c

```diff
@@ -249,50 +242,34 @@ void Impl_GC7_sweep1(uint8_t *g, uint64_t *h_index, uint64_t limit) {
 }
 */
 
-void Impl_GC7_sweep_body_helper_with_free_list(uint8_t *g, uint64_t *v_index,
-                                               uint64_t *free_list_ptr) {
-  uint64_t v_index_val = *v_index;
-  uint64_t c =
-      Impl_GC_closure_infix_color_of_block((uint64_t)(Hp_val(v_index_val)), g);
-
-  if (Impl_GC_closure_infix_wosize_of_block((uint64_t)Hp_val(v_index_val), g) ==
-      0) {
-    /* printf("Empty block %p with color: %d\n", (uint64_t *)v_index_val, c); */
-    return;
-  }
+void Impl_GC_closure_infix_sweep_body_helper_with_free_list1(
+    uint8_t *g, uint64_t *f_index, uint64_t *free_list_ptr) {
+  uint64_t f_index_val = *f_index;
+  uint64_t h_index = Spec_GC_closure_infix_hd_address(f_index_val);
+  uint64_t c = Impl_GC_closure_infix_color_of_block(h_index, g);
+  uint64_t wz = Impl_GC_closure_infix_wosize_of_block(h_index, g);
```

Comment:
```c
if(wz==0){
   return;
}
```

This is needed because we don't want to deal with 0 sized objects here in sweep(we ignore them in other places too so they should be ignored here as well). In verification,we're assume no 0 sized objects I believe but in reality it's not the case, so this change in verified code is needed. This was causing a crash when I was trying to run locally